homeassistant/components/fritzbox/config_flow.py

```diff
@@ -165,16 +175,20 @@ async def async_step_confirm(self, user_input=None):
             errors=errors,
         )
 
-    async def async_step_reauth(self, data):
+    async def async_step_reauth(self, data: dict[str, str]) -> FlowResult:
         """Trigger a reauthentication flow."""
-        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        assert isinstance(entry, ConfigEntry)
```

Comment:
I think you meant this.
```suggestion
        assert entry is not None
```